src/NuGet.Clients/NuGet.VisualStudio.Common/SolutionRestore/SolutionRestoreRequest.cs

```diff
@@ -22,14 +24,25 @@ public sealed class SolutionRestoreRequest
 
         public RestoreOperationSource RestoreSource { get; }
 
+        public ExplicitRestoreReason ExplicitRestoreReason { get; }
+
         public Guid OperationId => Guid.NewGuid();
 
         public SolutionRestoreRequest(
             bool forceRestore,
             RestoreOperationSource restoreSource)
+            : this(forceRestore, restoreSource, ExplicitRestoreReason.None)
+        {
+        }
+
+        public SolutionRestoreRequest(
+            bool forceRestore,
+            RestoreOperationSource restoreSource,
```

Comment:
nit: I wonder if we should check for RestoreOperationSource and if it's explicit, whether we should actually throw and disallow ExplictRestoreReason to be None in those cases. 

Happy to chat more, but comfortable approving as is.